syntax = "proto2";
package et;


message TerminalInfo {
  optional string id = 1;
  optional int32 row = 2;
  optional int32 column = 3;
  optional int32 width = 4;
  optional int32 height = 5;
}

message PortForwardSourceRequest {
  optional int32 sourceport = 1;
  optional int32 destinationport = 2;
}

message PortForwardDestinationRequest {
  optional int32 port = 1;
  optional int32 fd = 2;
}

message PortForwardDestinationResponse {
  optional int32 clientfd = 1;
  optional int32 socketid = 2;
  optional string error = 3;
}

message PortForwardData {
  optional bool sourcetodestination = 1;
  optional int32 socketid = 2;
  optional bytes buffer = 3;
  optional string error = 4;
  optional bool closed = 5;
}

message InitialPayload {
  optional bool jumphost = 1 [default = false];
}

